pandapower/plotting/plotly/traces.py

```diff
@@ -480,14 +480,15 @@ def _create_branch_trace(net, branches=None, use_branch_geodata=True, respect_se
         else:
             raise NotImplementedError("respect separtors is only implements for switches, "
                                       "not for {}s.".format(separator_element))
-    branches_to_plot = net[branch_element].loc[set(net[branch_element].index) & (set(branches) - no_go_branches)]
+    branches_to_plot = net[branch_element].loc[list(set(net[branch_element].index) &
+                                                    (set(branches) - no_go_branches))]
     no_go_branches_to_plot = None
     branch_geodata = branch_element + "_geodata"
     node_geodata = node_element + "_geodata"
     use_branch_geodata = use_branch_geodata if net[branch_geodata].shape[0] > 0 else False
     if use_branch_geodata:
-        branches_to_plot = branches_to_plot.loc[set(branches_to_plot.index) &
-                                                set(net[branch_geodata].index)]
+        branches_to_plot = branches_to_plot.loc[list(set(branches_to_plot.index) &
```

Comment:
same as above, I think np.intersect1d is cleaner

Reply:
But only if it comes from arrays/lists. If branches_to_plot is set from the beginning, than it is better to use set operations from Python (as in the code here).

Reply:
oh, I see that the index is cast to set first, so I think numpy would be faster here